picky-asn1-der/src/application_tag.rs

```diff
@@ -0,0 +1,129 @@
+use crate::misc::Length;
+use crate::Asn1RawDer;
+use picky_asn1::tag::{Tag, TagPeeker};
+use serde::de::{Error, SeqAccess};
+use serde::{de, ser};
+use std::fmt;
+use std::fmt::Debug;
+
+#[derive(Debug, PartialEq)]
+pub struct ApplicationTag<V: Debug + PartialEq, const T: u8>(V);
+
+impl<V: Debug + PartialEq, const T: u8> ApplicationTag<V, T> {
+    pub fn from(value: V) -> Self {
+        Self(value)
+    }
+}
+
+impl<'de, V: de::Deserialize<'de> + Debug + PartialEq, const T: u8> de::Deserialize<'de> for ApplicationTag<V, T> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        struct Visitor<E>(Option<E>, u8);
+
+        impl<E> Visitor<E> {
+            pub fn new(tag: u8) -> Self {
+                Self(None, tag)
+            }
+        }
+
+        impl<'de, E: de::Deserialize<'de> + Debug + PartialEq> de::Visitor<'de> for Visitor<E> {
+            type Value = E;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str(&format!("A valid DER-encoded ApplicationTag{}", self.1))
```

Comment:
```suggestion
                formatter.write_str(&format!("A valid DER-encoded ApplicationTag{:02}", self.1))
```
To print the leading zero if applicable